volatility3/framework/plugins/linux/tracing/ftrace.py

```diff
@@ -0,0 +1,316 @@
+        ):
+            vollog.info(
+                "A callback module origin could not be determined. hidden_modules plugin will be run to detect additional modules.",
+            )
+            known_modules_addresses = set(
+                kernel_layer.canonicalize(module.vol.offset)
+                for module in modxview.Modxview.flatten_run_modules_results(
+                    known_modules
+                )
+            )
+            modules_memory_boundaries = (
+                hidden_modules.Hidden_modules.get_modules_memory_boundaries(
+                    context, kernel_name
+                )
+            )
+            known_modules["hidden_modules"] = list(
+                hidden_modules.Hidden_modules.get_hidden_modules(
+                    context,
+                    kernel_name,
+                    known_modules_addresses,
+                    modules_memory_boundaries,
+                )
+            )
+            # Lookup the updated list to see if hidden_modules was able
+            # to find the missing module
+            module = linux_utilities_modules.Modules.module_lookup_by_address(
+                context,
+                kernel.layer_name,
+                modxview.Modxview.flatten_run_modules_results(known_modules),
+                callback,
+            )
+
+        # Fetch more information about the module
+        if module is not None:
+            module_address = module.vol.offset
+            module_name = module.get_name()
+            callback_symbol = module.get_symbol_by_address(callback)
+        else:
+            vollog.warning(
+                f"Could not determine ftrace_ops@{ftrace_ops.vol.offset:#x} callback {callback:#x} module origin.",
+            )
+
+        # Iterate over ftrace_func_entry list
+        for ftrace_func_entry in cls.extract_hash_table_filters(ftrace_ops):
+            hook_address = ftrace_func_entry.ip.cast("pointer")
+
+            # Determine the symbols associated with a hook
+            hooked_symbols = kernel.get_symbols_by_absolute_location(hook_address)
+            hooked_symbols = ",".join(
+                [
+                    hooked_symbol.split(constants.BANG)[-1]
+                    for hooked_symbol in hooked_symbols
+                ]
+            )
+            yield ParsedFtraceOps(
+                ftrace_ops.vol.offset,
+                callback_symbol,
+                callback,
+                hooked_symbols,
+                module_name,
+                module_address,
+                # FtraceOpsFlags(ftrace_ops.flags).name is valid in > Python3.10, but
+                # returns None <= Python 3.10. We need to manipulate it like so to ensure compatibility:
+                # FtraceOpsFlags.FTRACE_OPS_FL_IPMODIFY|FTRACE_OPS_FL_ALLOC_TRAMP
+                # -> FTRACE_OPS_FL_IPMODIFY,FTRACE_OPS_FL_ALLOC_TRAMP
+                str(FtraceOpsFlags(ftrace_ops.flags)).split(".")[-1].replace("|", ","),
```

Comment:
This... seems nasty.  Any chance you could do it in more reable pieces before the yield statement please?